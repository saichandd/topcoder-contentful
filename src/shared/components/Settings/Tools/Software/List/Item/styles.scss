@import "~styles/mixins";

.container {
  display: flex;
  flex-direction: row;
  justify-items: center;
  align-items: center;
  justify-content: space-between;
  height: 100%;
  width: 100%;
  padding: 20px;
  border: 1px solid $tc-gray-10;
  border-bottom: none;
}

.software-info {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
}

.software-icon {
  height: 49px;
  width: 49px;
  margin-right: 18px;
}

.software-parameters {
  @include roboto-medium;

  display: flex;
  flex-direction: column;
  justify-items: center;
  justify-content: left;
  font-size: 15px;
  line-height: 20px;
  text-transform: capitalize;
}

.parameter-first-line {
  font-weight: 500;
  color: $tc-black;
  margin-bottom: 10px;
}

.parameter-second-line {
  color: $tc-gray-50;
  font-weight: 400;
}

.delete {
  display: flex;
  flex-direction: column;
  justify-items: center;
  justify-content: center;
  cursor: pointer;
  outline-style: none;

  img {
    margin-bottom: 10px;

    @include xs-to-sm {
      margin-bottom: 0;
    }
  }

  p {
    @include roboto-regular;

    font-size: 11px;
    line-height: 15px;
    font-weight: 400;
    color: $tc-gray-50;

    @include xs-to-sm {
      display: none;
    }
  }
}
