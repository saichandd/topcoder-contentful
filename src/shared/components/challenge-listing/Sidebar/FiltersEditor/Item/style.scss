@import "~styles/mixins";

.errorMsg {
  color: $tc-red;
  line-height: 13px;
  margin-bottom: 6px;
}

.icon-arrows-v {
  height: 13px;
  left: $base-unit;
  position: absolute;
  fill: $tc-dark-blue;
  top: 12px;
}

.icon-cross {
  fill: $tc-dark-blue;
  height: 13px;
  position: relative;
  top: 2px;
}

.cross-tooltip {
  color: $tc-white;
  border-radius: 3px;
  border: 1px solid $tc-gray-80;
  background: $tc-gray-80;
  display: none;
  line-height: 13px;
  padding: $base-unit * 3;
  position: absolute;
  right: 0;
  top: -50px;
  z-index: 10;

  &::before {
    content: '';
    position: absolute;
    display: inline-block;
    width: 0;
    height: 0;
    border-color: transparent;
    border-style: solid;
    bottom: -6px;
    margin-left: -5px;
    border-width: 5px 5px 0;
    left: 67%;
    border-top-color: $tc-gray-80;
  }
}

.ActiveFilterItem {
  cursor: pointer;
  margin: 0 -2 * $base-unit;
  padding: 0 5 * $base-unit;
  position: relative;

  .input {
    background: transparent;
    border: none;
    display: inline-block;
    height: 13px;
    font-size: 13px;
    margin-bottom: 0;
    padding: 2 * $base-unit 0;
    outline: none;
    width: 270px;

    &.error {
      border: 1px solid $tc-red;
    }

    &:hover {
      box-shadow: inset 0 0 2px 0 #b7b7b7;
    }
  }

  .icon-arrows-v {
    display: none;
  }

  .icon-cross {
    display: none;
  }

  .right {
    float: right;
  }
}

.ActiveFilterItem:hover {
  background: $tc-dark-blue-10;

  .icon-arrows-v {
    display: block;
  }

  .icon-cross {
    display: inline;
  }

  .right:hover {
    .cross-tooltip {
      display: block;
    }

    .icon-cross {
      background: $tc-dark-blue;
      border-radius: 26px;
      fill: $tc-white;
    }
  }
}
