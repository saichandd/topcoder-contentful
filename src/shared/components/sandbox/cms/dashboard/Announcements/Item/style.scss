@import "~styles/mixins";

.container {
  @include tc-body-md;

  background: white;
  border: 1px solid $tc-purple;
  padding: 10px;
  position: absolute;
  overflow: hidden;

  &:hover {
    background: $tc-purple-30;
    cursor: pointer;
  }
}

.title {
  @include tc-heading-md;

  white-space: nowrap;
}
