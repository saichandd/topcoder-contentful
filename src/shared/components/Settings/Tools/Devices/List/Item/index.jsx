/**
 * render device item.
 */
import React from 'react';
import PT from 'prop-types';
import ReactSVG from 'react-svg';
import { isomorphy } from 'topcoder-react-utils';

import './styles.scss';

let assets;
if (isomorphy.isClientSide()) {
  assets = require.context('assets/images/tools/device-types', false, /svg/);
}

export default function Item(props) {
  const {
    device,
    index,
    onDeleteItem,
  } = props;

  return (
    <div styleName="container">
      <div styleName="device-info">
        <div styleName="device-icon">
          { assets && assets.keys().includes(`./${device.deviceType.toLowerCase()}.svg`) ? <ReactSVG path={assets(`./${device.deviceType.toLowerCase()}.svg`)} /> : '' }
        </div>
        <div styleName="device-parameters">
          <div styleName="parameter-first-line">
            {
              `${device.manufacturer} | ${device.model} | ${device.deviceType}`
            }
          </div>
          <div styleName="parameter-second-line">
            {
              `${device.operatingSystem} ${device.osVersion} ${device.osLanguage}`
            }
          </div>
        </div>
      </div>
      <a
        tabIndex={0}
        styleName="delete"
        role="button"
        onKeyPress={() => onDeleteItem(index)}
        onClick={() => onDeleteItem(index)}
      >
        <img src={assets('./trash.svg')} alt="delete-icon" />
        <p>
Delete
        </p>
      </a>
    </div>
  );
}

Item.propTypes = {
  device: PT.shape().isRequired,
  index: PT.number.isRequired,
  onDeleteItem: PT.func.isRequired,
};
