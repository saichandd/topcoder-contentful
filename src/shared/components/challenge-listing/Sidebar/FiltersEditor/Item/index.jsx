import React from 'react';
import PT from 'prop-types';

import './style.scss';

import ArrowsMoveVertical from '../../../Icons/ArrowsMoveVertical';
import UiSimpleRemove from '../../../Icons/ui-simple-remove.svg';

export default function Item({
  error,
  changeFilterName,
  deleteSavedFilter,
  dragSavedFilterMove,
  dragSavedFilterStart,
  name,
  resetFilterName,
  updateSavedFilter,
}) {
  return (
    <div
      styleName="ActiveFilterItem"
      draggable
      onDrag={event => dragSavedFilterMove(event)}
      onDragStart={event => dragSavedFilterStart(event)}
    >
      <ArrowsMoveVertical styleName="icon-arrows-v" />
      <input
        styleName={`input ${error ? 'error' : ''}`}
        onBlur={() => !error && updateSavedFilter()}
        onChange={event => changeFilterName(event.target.value)}
        onKeyDown={(event) => {
          switch (event.key) {
            case 'Enter': return event.target.blur();
            case 'Escape': {
              event.target.blur();
              return resetFilterName();
            }
            default: return undefined;
          }
        }}
        value={name}
        type="text"
      />
      <span
        onClick={deleteSavedFilter}
        onKeyPress={deleteSavedFilter}
        role="button"
        styleName="right"
        tabIndex={0}
      >
        <UiSimpleRemove styleName="icon-cross" />
        <div styleName="cross-tooltip">
Delete Filter
        </div>
      </span>
      { Boolean(error) && (
        <div styleName="errorMsg">
          {error}
        </div>
      )}
    </div>
  );
}

Item.defaultProps = {
  error: '',
};

Item.propTypes = {
  deleteSavedFilter: PT.func.isRequired,
  dragSavedFilterMove: PT.func.isRequired,
  dragSavedFilterStart: PT.func.isRequired,
  error: PT.string,
  name: PT.string.isRequired,
  changeFilterName: PT.func.isRequired,
  resetFilterName: PT.func.isRequired,
  updateSavedFilter: PT.func.isRequired,
};
