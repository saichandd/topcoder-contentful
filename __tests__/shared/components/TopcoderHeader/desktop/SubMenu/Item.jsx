import React from 'react';
import Renderer from 'react-test-renderer/shallow';
import Item from 'components/TopcoderHeader/desktop/SubMenu/Item';

test('Matches shallow shapshot', () => {
  const renderer = new Renderer();

  renderer.render((
    <Item
      currentSubMenuTitle="title"
      icon="icon url"
      title="title"
      link="menu url"
    />
  ));
  expect(renderer.getRenderOutput()).toMatchSnapshot();

  renderer.render((
    <Item
      currentSubMenuTitle="another title"
      icon="icon url"
      title="title"
      link="menu url"
    />
  ));
  expect(renderer.getRenderOutput()).toMatchSnapshot();
});
