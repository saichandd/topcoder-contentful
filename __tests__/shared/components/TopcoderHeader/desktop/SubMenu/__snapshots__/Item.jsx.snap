// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches shallow shapshot 1`] = `
<li
  className="src-shared-components-TopcoderHeader-desktop-SubMenu-Item-___style__item___2Q-ud src-shared-components-TopcoderHeader-desktop-SubMenu-Item-___style__current___38PHh"
  role="button"
  tabIndex={0}
>
  <Link
    enforceA={false}
    to="menu url"
  >
    icon url
    title
  </Link>
</li>
`;

exports[`Matches shallow shapshot 2`] = `
<li
  className="src-shared-components-TopcoderHeader-desktop-SubMenu-Item-___style__item___2Q-ud"
  role="button"
  tabIndex={0}
>
  <Link
    enforceA={false}
    to="menu url"
  >
    icon url
    title
  </Link>
</li>
`;
